simplicity-sys/src/c_jets/c_env.rs

```diff
@@ -0,0 +1,233 @@
+use std::os::raw::{c_uchar, c_uint};
+
+use libc::size_t;
+
+/// Documentation of CRawInputData/CRawOutputData/CRawTapData/CRaw
+/// Data structure for holding data that CTransaction points to.
+///
+/// Why do we this special data structure?
+///     1. We need to keep the data in memory until the CTransaction is dropped.
+///     2. The memory is Transaction is not saved in the same format as required by FFI.
+/// We use more ergonomics in rust to allow better UX which interfere with the FFI. For example,
+/// the Value is stored as Tagged Union, but we require it to be a slice of bytes in elements format.
+///     3. Allocating inside FFI functions does not work because the memory is freed after the function returns.
+///     4. We only create allocations for data fields that are stored differently from the
+/// consensus serialization format.
+#[derive(Debug)]
+pub struct CRawOutputData {
+    pub asset: Vec<c_uchar>,
+    pub value: Vec<c_uchar>,
+    pub nonce: Vec<c_uchar>,
+    pub surjection_proof: Vec<c_uchar>,
+    pub range_proof: Vec<c_uchar>,
+}
```

Comment:
In 59c2942084530ec7c2d06b9147dccd3156aa51c3:

This structure (and others) shouldn't have the `C` prefix since it's not actually a FFI type. I think we should only use `C` for things with `repr(C)` on them.